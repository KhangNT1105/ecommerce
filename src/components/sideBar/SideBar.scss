@import "../../styles/variables";

.layout-sidebar {
  .sidebar {
    h6 {
      color: $drop-down-grey-color;
      font-weight: bold;
      text-transform: uppercase;
    }

    &.custom-sidebar-slim {
      width: auto;
      height: 100%;
      min-width: 60px;

      i.fa {
        margin: 0;
      }

      ul.sidebar-menu {
        margin-bottom: 0;
      }
    }

    .logo-slim {
      width: 20px;
    }

    &-submenu,
    &-menu {
      & li {
        & a:focus {
          outline: none;
        }
      }
    }
  }

  small.small {
    color: $white;
  }
}
