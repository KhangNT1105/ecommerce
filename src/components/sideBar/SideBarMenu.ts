import { ENTITIES, API_ENTITIES_NAME } from 'constants/enum';

export const mockDataSidebar = {
  menus: [
    {
      title: 'SIDEBAR_MENU.CONTENT',
      featureName: 'Content',
      icon: 'fa-youtube-play',
      subMenus: [
        {
          icon: 'fa-film',
          title: 'SIDEBAR_MENU.SHOWS',
          featureName: 'Shows',
          to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.SHOWS}`,
          menuItemName: `${API_ENTITIES_NAME.SHOWS}`,
          parent: true,
          entity: `${ENTITIES.NAME.SHOWS}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.SHOWS',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.SHOWS}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.SHOWS',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.SHOWS}/create`
            }
          ]
        },
        {
          icon: 'fa-calendar-check-o',
          title: 'SIDEBAR_MENU.SCHEDULES',
          featureName: 'Schedules',
          to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.SCHEDULES}`,
          menuItemName: `${API_ENTITIES_NAME.SCHEDULES}`,
          parent: true,
          entity: `${ENTITIES.NAME.SCHEDULES}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.SCHEDULES',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.SCHEDULES}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.SCHEDULES',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.SCHEDULES}/create`
            }
          ]
        },
        {
          icon: 'fa-television',
          title: 'SIDEBAR_MENU.CHANNELS',
          featureName: 'Channels',
          to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CHANNELS}`,
          menuItemName: `${API_ENTITIES_NAME.CHANNELS}`,
          parent: true,
          entity: `${ENTITIES.NAME.CHANNELS}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.CHANNELS',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CHANNELS}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.CHANNELS',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CHANNELS}/create`
            }
          ]
        },
        {
          icon: 'fa-tags',
          title: 'SIDEBAR_MENU.CATALOGS',
          featureName: 'Catalogs',
          to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CATALOGS}`,
          menuItemName: `${API_ENTITIES_NAME.CATALOGS}`,
          parent: true,
          entity: `${ENTITIES.NAME.CATALOGS}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.CATALOGS',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CATALOGS}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.CATALOGS',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CATALOGS}/create`
            }
          ]
        },
        {
          icon: 'fa-th-large',
          title: 'SIDEBAR_MENU.CATEGORIES',
          featureName: 'Categories',
          to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CATEGORIES}`,
          menuItemName: `${API_ENTITIES_NAME.CATEGORIES}`,
          parent: true,
          entity: `${ENTITIES.NAME.CATEGORIES}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.CATEGORIES',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CATEGORIES}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.CATEGORIES',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CATEGORIES}/create`
            }
          ]
        },
        {
          icon: 'fa-user-secret',
          title: 'SIDEBAR_MENU.CASTINGS',
          featureName: 'Casts',
          to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CASTINGS}`,
          menuItemName: `${API_ENTITIES_NAME.CASTINGS}`,
          parent: true,
          entity: `${ENTITIES.NAME.CASTINGS}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.CASTS',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CASTINGS}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.CASTS',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CASTINGS}/create`
            }
          ]
        },
        {
          icon: 'fa-video-camera',
          title: 'SIDEBAR_MENU.VIDEO_CLIPS',
          featureName: 'VideoClips',
          to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CLIPS}`,
          menuItemName: `${API_ENTITIES_NAME.CLIPS}`,
          parent: true,
          entity: `${ENTITIES.NAME.CLIPS}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.VIDEO_CLIPS',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CLIPS}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.VIDEO_CLIPS',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.CLIPS}/create`
            }
          ]
        },
        {
          icon: 'fa-list-alt',
          title: 'SIDEBAR_MENU.VIDEO_PLAYLISTS',
          featureName: 'Video Playlists',
          to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.PLAYLISTS}`,
          menuItemName: `${API_ENTITIES_NAME.PLAYLISTS}`,
          parent: true,
          entity: `${ENTITIES.NAME.PLAYLISTS}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.VIDEO_PLAYLISTS',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.PLAYLISTS}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.VIDEO_PLAYLISTS',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.PLAYLISTS}/create`
            }
          ]
        },
        {
          icon: 'fa-refresh',
          title: 'SIDEBAR_MENU.UPDATES',
          featureName: 'Updates',
          to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.UPDATES}`,
          menuItemName: `${API_ENTITIES_NAME.UPDATES}`,
          parent: true,
          entity: `${ENTITIES.NAME.UPDATES}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.UPDATES',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.UPDATES}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.UPDATES',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.CONTENT}/${ENTITIES.NAME.UPDATES}/create`
            }
          ]
        }
      ]
    },
    {
      title: 'SIDEBAR_MENU.LICENSES',
      featureName: 'Licenses',
      icon: 'fa-paperclip',
      subMenus: [
        {
          icon: 'fa-suitcase',
          title: 'SIDEBAR_MENU.CONTRACTS',
          featureName: 'Contracts',
          to: `/${ENTITIES.PATH.LICENSES}/${ENTITIES.NAME.CONTRACTS}`,
          menuItemName: `${API_ENTITIES_NAME.CONTRACTS}`,
          parent: true,
          entity: `${ENTITIES.NAME.CONTRACTS}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.CONTRACTS',
              to: `/${ENTITIES.PATH.LICENSES}/${ENTITIES.NAME.CONTRACTS}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.CONTRACTS',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.LICENSES}/${ENTITIES.NAME.CONTRACTS}/create`
            }
          ]
        },
        {
          icon: 'fa-male',
          title: 'SIDEBAR_MENU.LICENSORS',
          featureName: 'Licensors',
          to: `/${ENTITIES.PATH.LICENSES}/${ENTITIES.NAME.LICENSORS}`,
          menuItemName: `${API_ENTITIES_NAME.LICENSORS}`,
          parent: true,
          entity: `${ENTITIES.NAME.LICENSORS}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.LICENSORS',
              to: `/${ENTITIES.PATH.LICENSES}/${ENTITIES.NAME.LICENSORS}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.LICENSORS',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.LICENSES}/${ENTITIES.NAME.LICENSORS}/create`
            }
          ]
        }
      ]
    },
    {
      title: 'SIDEBAR_MENU.USER_MANAGEMENT',
      featureName: 'UserManagement',
      icon: 'fa-user-plus',
      subMenus: [
        {
          icon: 'fa-user',
          title: 'SIDEBAR_MENU.SYSTEM_USERS',
          featureName: 'SystemUsers',
          to: `/${ENTITIES.PATH.USER_MANAGEMENT}/${ENTITIES.NAME.SYSTEM_USERS}`,
          menuItemName: `${API_ENTITIES_NAME.SYSTEM_USERS}`,
          parent: true,
          entity: `${ENTITIES.NAME.SYSTEM_USERS}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.SYSTEM_USERS',
              to: `/${ENTITIES.PATH.USER_MANAGEMENT}/${ENTITIES.NAME.SYSTEM_USERS}`
            }
          ]
        },
        {
          icon: 'fa-users',
          title: 'SIDEBAR_MENU.USER_GROUPS',
          featureName: 'UserGroups',
          to: `/${ENTITIES.PATH.USER_MANAGEMENT}/${ENTITIES.NAME.USER_GROUPS}`,
          menuItemName: `${API_ENTITIES_NAME.USER_GROUPS}`,
          parent: true,
          entity: `${ENTITIES.NAME.USER_GROUPS}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.USER_GROUPS',
              to: `/${ENTITIES.PATH.USER_MANAGEMENT}/${ENTITIES.NAME.USER_GROUPS}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.USER_GROUPS',
              name: 'btnCreate',
              to: `/${ENTITIES.PATH.USER_MANAGEMENT}/${ENTITIES.NAME.USER_GROUPS}/create`
            }
          ]
        }
      ]
    },
    {
      title: 'SIDEBAR_MENU.SYSTEM_CONFIGURATION',
      featureName: 'SystemConfiguration',
      icon: 'fa-cogs',
      subMenus: [
        {
          icon: 'fa-credit-card',
          title: 'SIDEBAR_MENU.PAYMENT_GATEWAYS',
          featureName: 'PaymentGateways',
          to: `/${ENTITIES.PATH.SYSTEM_CONFIGURATION}/${ENTITIES.NAME.PAYMENT_GATEWAYS}`,
          menuItemName: `${API_ENTITIES_NAME.CONFIGURATIONS}`,
          parent: true,
          entity: `${ENTITIES.NAME.CONFIGURATIONS}`,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.PAYMENT_GATEWAYS',
              to: `/${ENTITIES.PATH.SYSTEM_CONFIGURATION}/${ENTITIES.NAME.PAYMENT_GATEWAYS}`
            }
          ]
        },
        {
          icon: 'fa-shield',
          title: 'SIDEBAR_MENU.RISK_ENGINE_RULES_CONFIG',
          featureName: 'PaymentGateways',
          to: `/${ENTITIES.PATH.SYSTEM_CONFIGURATION}/${ENTITIES.NAME.RISK_ENGINE_RULES_CONFIG}`,
          menuItemName: `${API_ENTITIES_NAME.CONFIGURATIONS}`,
          parent: true,
          subMenus: [
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.ALL.RISK_ENGINE_RULES_CONFIG',
              to: `/${ENTITIES.PATH.SYSTEM_CONFIGURATION}/${ENTITIES.NAME.RISK_ENGINE_RULES_CONFIG}`
            },
            {
              icon: 'd-none',
              title: 'SIDEBAR_MENU.CREATE.RISK_ENGINE_RULES_CONFIG',
              to: `/${ENTITIES.PATH.SYSTEM_CONFIGURATION}/${ENTITIES.NAME.RISK_ENGINE_RULES_CONFIG}/create`
            }
          ]
        }
      ]
    }
  ]
};

export default {
  mockDataSidebar
};
