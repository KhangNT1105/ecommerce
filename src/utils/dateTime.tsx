import {
  format,
  parse,
  getTime,
  subDays,
  subMonths,
  startOfWeek,
  endOfWeek,
  startOfMonth,
  endOfMonth,
  startOfToday,
  endOfToday,
  startOfDay,
  endOfDay
} from 'date-fns';

import { DATE_FORMAT, DATE_TIME_FULL_FORMAT } from 'constants/index';

export type DateRangeType = { from: string | null; to: string | null };

export const formatDate = (timestamp: string) => {
  return format(parse(timestamp, 'T', new Date()), DATE_FORMAT);
};

export type DateRangeProps = {
  dateFrom: string | null;
  dateTo: string | null;
};

/**
 * get timestamp from a string
 * @param: value string
 */
export const getTimestamp = (value: string): number => {
  return getTime(parse(value, DATE_FORMAT, new Date()));
};

export const getUTCDateFn = (date: Date) => {
  return new Date(
    date.getUTCFullYear(),
    date.getUTCMonth(),
    date.getUTCDate(),
    date.getUTCHours(),
    date.getUTCMinutes(),
    date.getUTCSeconds()
  );
};

/**
 * Description: combine day from DatePicker and local time then convert to UTC Date
 * @param date: from date picker
 * @return: UTC Date
 */
export const convertDatePickerToUTC = (date: Date) => {
  const today = new Date();
  const combineDateTime = new Date(
    date.getFullYear(),
    date.getMonth(),
    date.getDate(),
    today.getHours(),
    today.getMinutes(),
    today.getSeconds()
  );

  return getUTCDateFn(combineDateTime);
};

/**
 * Description: convert HH:MM:SS string to seconds
 * @param dateString
 * @param second
 */
export const convertTimeStringToMillisecond = (dateString: String) => {
  return (dateString as any).split(':').reduce((acc: any, time: any) => 60 * acc + +time) * 1000;
};

export const convertDateFromToUTC = (date: Date) => {
  const combineDateTime = new Date(date.getFullYear(), date.getMonth(), date.getDate());

  return getUTCDateFn(startOfDay(combineDateTime));
};

export const convertDateToToUTC = (date: Date) => {
  const combineDateTime = new Date(date.getFullYear(), date.getMonth(), date.getDate());

  return getUTCDateFn(endOfDay(combineDateTime));
};

export const getToday = (): DateRangeProps => {
  const dateFrom = getUTCDateFn(startOfToday());
  const dateTo = getUTCDateFn(endOfToday());

  return {
    dateFrom: format(dateFrom, DATE_TIME_FULL_FORMAT),
    dateTo: format(dateTo, DATE_TIME_FULL_FORMAT)
  };
};

export const getLast7days = (): DateRangeProps => {
  const dateTo = format(getUTCDateFn(endOfToday()), DATE_TIME_FULL_FORMAT);
  const dateFrom = format(getUTCDateFn(subDays(startOfToday(), 7)), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLast30days = (): DateRangeProps => {
  const dateTo = format(getUTCDateFn(endOfToday()), DATE_TIME_FULL_FORMAT);
  const dateFrom = format(getUTCDateFn(subDays(startOfToday(), 30)), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLastWeek = (): DateRangeProps => {
  const dateFrom = format(getUTCDateFn(startOfWeek(subDays(startOfToday(), 7))), DATE_TIME_FULL_FORMAT);
  const dateTo = format(getUTCDateFn(endOfWeek(subDays(endOfToday(), 7))), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLastMonth = (): DateRangeProps => {
  const dateFrom = format(getUTCDateFn(startOfMonth(subMonths(startOfToday(), 1))), DATE_TIME_FULL_FORMAT);
  const dateTo = format(getUTCDateFn(endOfMonth(subMonths(endOfToday(), 1))), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const convertMillisecondToHHMMSS = (milliseconds: number) => {
  if (!milliseconds) return '';
  const date = new Date(milliseconds);
  const hh = date.getUTCHours();
  const mm = date.getUTCMinutes();
  const ss = date.getSeconds();
  const hours = hh < 10 ? `0${hh}` : `${hh}`;
  const minutes = mm < 10 ? `0${mm}` : `${mm}`;
  const seconds = ss < 10 ? `0${ss}` : `${ss}`;

  return `${hours}:${minutes}:${seconds}`;
};
