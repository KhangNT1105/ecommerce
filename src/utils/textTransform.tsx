export const upperCaseFirst = (str: String = '') => {
  return str.toLowerCase().replace(/^\w/, (chars) => chars.toUpperCase());
};
