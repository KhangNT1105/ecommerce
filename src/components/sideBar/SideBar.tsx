import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import SideBarMiddleNav from './sideBarMiddleNav/SideBarMiddleNav';
import { SideBarSection } from './sideBarSection/SideBarSection';
import { useTranslation } from 'react-i18next';
import { SideBarProps, SideBarState } from './SideBar.d';

import './SideBar.scss';

const SideBar: React.FC<SideBarProps> = ({ onClose, collapsed, slim, animationsDisabled, menus, isNoPermission }) => {
  const { t } = useTranslation();

  const [state] = useState<SideBarState>({
    entryAnimationFinished: true
  });

  const { entryAnimationFinished } = state;

  const classes = classNames('sidebar', 'sidebar--animations-enabled', {
    'sidebar--slim custom-sidebar-slim': slim,
    'sidebar--collapsed': collapsed,
    'sidebar--animations-disabled': animationsDisabled,
    'sidebar--animate-entry-complete': entryAnimationFinished
  });

  const handleCloseSideBar = () => {
    onClose && onClose();
  };

  return (
    <div className={classes}>
      <div className="sidebar__close" onClick={handleCloseSideBar}>
        <i className="fa fa-times-circle fa-fw" />
      </div>
      <SideBarSection>
        <div className="sidebar__brand">
          <Link to="/">
            <img src={''} className={`logo-slim ${slim ? '' : 'd-none'}`} alt={t('ACEXIS_LOGO_ALT')} />
            <img src={''} className={`${slim ? 'd-none' : ''}`} alt={t('ACEXIS_LOGO_ALT')} />
            {!slim && (
              <small className="d-block small">
                <strong>{t('CMS_TITLE')}</strong>
              </small>
            )}
          </Link>
        </div>
      </SideBarSection>
      {!isNoPermission && (
        <div className="sidebar__mobile-fluid">
          <SideBarSection fluid={true} cover={true}>
            <SideBarMiddleNav menus={menus} slim={slim} />
          </SideBarSection>
        </div>
      )}
    </div>
  );
};

export default SideBar;
