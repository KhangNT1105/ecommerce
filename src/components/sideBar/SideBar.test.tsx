import React from 'react';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import { testSnapshots } from '../../utils/test';
import SideBar from './SideBar';
import { mockDataSidebar } from './SideBarMenu';

let defaultProps: any = mockDataSidebar;
const wrapper = (props: any) => (
  <BrowserRouter>
    <SideBar {...props} />
  </BrowserRouter>
);

describe('<SideBar />', () => {
  testSnapshots(wrapper, [
    {
      props: defaultProps,
      description: 'default render'
    }
  ]);
});

describe('Test Logic', () => {
  test('should close side bar when click close icon', () => {
    defaultProps = {
      ...defaultProps,
      onClose: jest.fn(),
      isNoPermission: false
    };
    const wrapper = shallow(<SideBar {...defaultProps} />);
    wrapper.find('.sidebar__close').simulate('click');
    expect(defaultProps.onClose).toHaveBeenCalled();
  });
});
