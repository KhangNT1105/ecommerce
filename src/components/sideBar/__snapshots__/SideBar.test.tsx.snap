// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> snapshots default render 1`] = `
<div
  className="sidebar sidebar--animations-enabled sidebar--animate-entry-complete"
>
  <div
    className="sidebar__close"
    onClick={[Function]}
  >
    <i
      className="fa fa-times-circle fa-fw"
    />
  </div>
  <div
    className="sidebar__section"
  >
    <div
      className="sidebar__brand"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="ACEXIS_LOGO_ALT"
          className="logo-slim d-none"
          src=""
        />
        <img
          alt="ACEXIS_LOGO_ALT"
          className=""
          src=""
        />
        <small
          className="d-block small"
        >
          <strong>
            CMS_TITLE
          </strong>
        </small>
      </a>
    </div>
  </div>
  <div
    className="sidebar__mobile-fluid"
  >
    <div
      className="sidebar__section sidebar__section--fluid sidebar__section--cover"
    >
      <h6
        className="pl-2"
      >
        <i
          className="fa fa-fw fa-youtube-play mr-2"
        />
        SIDEBAR_MENU.CONTENT
      </h6>
      <ul
        className="sidebar-menu"
      >
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-film"
            />
            <span>
              SIDEBAR_MENU.SHOWS
            </span>
          </div>
        </li>
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-calendar-check-o"
            />
            <span>
              SIDEBAR_MENU.SCHEDULES
            </span>
          </div>
        </li>
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-television"
            />
            <span>
              SIDEBAR_MENU.CHANNELS
            </span>
          </div>
        </li>
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-tags"
            />
            <span>
              SIDEBAR_MENU.CATALOGS
            </span>
          </div>
        </li>
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-th-large"
            />
            <span>
              SIDEBAR_MENU.CATEGORIES
            </span>
          </div>
        </li>
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-user-secret"
            />
            <span>
              SIDEBAR_MENU.CASTINGS
            </span>
          </div>
        </li>
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-video-camera"
            />
            <span>
              SIDEBAR_MENU.VIDEO_CLIPS
            </span>
          </div>
        </li>
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-list-alt"
            />
            <span>
              SIDEBAR_MENU.VIDEO_PLAYLISTS
            </span>
          </div>
        </li>
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-refresh"
            />
            <span>
              SIDEBAR_MENU.UPDATES
            </span>
          </div>
        </li>
      </ul>
      <h6
        className="pl-2"
      >
        <i
          className="fa fa-fw fa-paperclip mr-2"
        />
        SIDEBAR_MENU.LICENSES
      </h6>
      <ul
        className="sidebar-menu"
      >
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-suitcase"
            />
            <span>
              SIDEBAR_MENU.CONTRACTS
            </span>
          </div>
        </li>
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-male"
            />
            <span>
              SIDEBAR_MENU.LICENSORS
            </span>
          </div>
        </li>
      </ul>
      <h6
        className="pl-2"
      >
        <i
          className="fa fa-fw fa-user-plus mr-2"
        />
        SIDEBAR_MENU.USER_MANAGEMENT
      </h6>
      <ul
        className="sidebar-menu"
      >
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-user"
            />
            <span>
              SIDEBAR_MENU.SYSTEM_USERS
            </span>
          </div>
        </li>
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-users"
            />
            <span>
              SIDEBAR_MENU.USER_GROUPS
            </span>
          </div>
        </li>
      </ul>
      <h6
        className="pl-2"
      >
        <i
          className="fa fa-fw fa-cogs mr-2"
        />
        SIDEBAR_MENU.SYSTEM_CONFIGURATION
      </h6>
      <ul
        className="sidebar-menu"
      >
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-credit-card"
            />
            <span>
              SIDEBAR_MENU.PAYMENT_GATEWAYS
            </span>
          </div>
        </li>
        <li
          className="sidebar-submenu__entry"
        >
          <div
            className="sidebar-submenu__entry__link undefined"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <i
              className="fa fa-fw fa-shield"
            />
            <span>
              SIDEBAR_MENU.RISK_ENGINE_RULES_CONFIG
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
